hir-analysis/src/analyses/spills.rs

```diff
@@ -2715,116 +2886,168 @@ mod tests {
+            let blk4 = b.create_block();
+            let blk5 = b.create_block();
+            let blk6 = b.create_block();
+
+            // entry -> block1(r, c, sum)
+            let r0 = b.u32(0, span);
+            let c0 = b.u32(0, span);
+            let s0 = b.u64(0, span);
+            b.br(blk1, [r0, c0, s0], span)?;
+
+            // block1: outer loop header
+            let r = b.append_block_param(blk1, Type::U32, span);
+            let c = b.append_block_param(blk1, Type::U32, span);
+            let sum = b.append_block_param(blk1, Type::U64, span);
+            b.switch_to_block(blk1);
+            let cond_outer = b.eq(r, v1, span)?;
+            Cf::cond_br(&mut b, cond_outer, blk2, [], blk3, [], span)?;
+
+            // block2: return sum
+            b.switch_to_block(blk2);
+            b.ret(Some(sum), span)?;
+
+            // block3: split edge to inner loop
+            b.switch_to_block(blk3);
+            b.br(blk4, [c, sum], span)?;
+
+            // block4: inner loop header
+            let col = b.append_block_param(blk4, Type::U32, span);
+            let acc = b.append_block_param(blk4, Type::U64, span);
+            b.switch_to_block(blk4);
+            let cond_inner = b.eq(col, v2, span)?;
+            // If inner loop finished (col == v2), forward state to blk5; otherwise run body
+            Cf::cond_br(&mut b, cond_inner, blk5, [col, acc], blk6, [], span)?;
+
+            // block5: latch to outer loop; receive forwarded inner state
+            b.switch_to_block(blk5);
+            let pcol = b.append_block_param(blk5, Type::U32, span);
+            let pacc = b.append_block_param(blk5, Type::U64, span);
+            let one = b.u32(1, span);
+            let r1 = b.add_unchecked(r, one, span)?;
+            b.br(blk1, [r1, pcol, pacc], span)?;
+
+            // block6: inner loop body
+            b.switch_to_block(blk6);
+            let one = b.u32(1, span);
+            let rowm1 = b.sub_unchecked(r, one, span)?;
+            let row_off = b.mul_unchecked(rowm1, v2, span)?;
+            let offset = b.add_unchecked(col, row_off, span)?;
+            let base = b.ptrtoint(v0, Type::U32, span)?;
+            let addr = b.add_unchecked(base, offset, span)?;
+            let ptr = b.inttoptr(
+                addr,
+                Type::Ptr(Arc::new(PointerType::new_with_address_space(
+                    Type::U64,
+                    AddressSpace::Element,
+                ))),
+                span,
+            )?;
+            let load = b.load(ptr, span)?;
+            let accn = b.add_unchecked(acc, load, span)?;
+            let one2 = b.u32(1, span);
+            let coln = b.add_unchecked(col, one2, span)?;
+            // Backedge: continue inner loop by jumping to its header
+            b.br(blk4, [coln, accn], span)?;
+
+            (blk1, blk3, blk4, blk5, v1)
+        };
 
-        // We expect a spill from block3 to block4, as due to operand stack pressure in the loop,
```

Comment:
@bitwalker I'm puzzled by the old `spills_loop_nest` test. It expected a spill (see removed asserts and comments) but in the new test I see max stack pressure at 9 and there are no spills. I recreated the IR from the old test (see doc comment) - check the `expect!` below.
Any ideas why the old test expected a spill?